src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -29,88 +27,74 @@ internal static class KeyAnalyzer
         /// operations, then we can select an ASCII-specific case-insensitive comparer which yields faster overall performance.
         /// </remarks>
         public static AnalysisResults Analyze(
-            ReadOnlySpan<string> uniqueStrings, bool ignoreCase, int minLength, int maxLength)
+            string[] uniqueStrings, bool ignoreCase, int minLength, int maxLength)
         {
-            Debug.Assert(!uniqueStrings.IsEmpty);
+            Debug.Assert(uniqueStrings.Length > 0);
 
-            // Try to pick a substring comparer. If we can't find a good substring comparer, fallback to a full string comparer.
-            AnalysisResults results;
-            if (minLength == 0 || !TryUseSubstring(uniqueStrings, ignoreCase, minLength, maxLength, out results))
+            if (minLength > 0)
             {
-                results = CreateAnalysisResults(uniqueStrings, ignoreCase, minLength, maxLength, 0, 0, static (s, _, _) => s.AsSpan());
-            }
-
-            return results;
-        }
+                const int MaxSubstringLengthLimit = 8; // arbitrary small-ish limit...it's not worth the increase in algorithmic complexity to analyze longer substrings
 
-        /// <summary>Try to find the minimal unique substring index/length to use for comparisons.</summary>
-        private static bool TryUseSubstring(ReadOnlySpan<string> uniqueStrings, bool ignoreCase, int minLength, int maxLength, out AnalysisResults results)
-        {
-            const int MaxSubstringLengthLimit = 8; // arbitrary small-ish limit... t's not worth the increase in algorithmic complexity to analyze longer substrings
+                // Sufficient uniqueness factor of 95% is good enough.
```

Comment:
Hoist the calculation of the number of acceptable collision out to the top level as it never actually changes for any of the strategy attempts (left/right and `count` of up to 8 characters... so as many at 15 excess calculations)